internal/controller/grafanaorganization_controller.go

```diff
@@ -131,24 +131,64 @@ func (r GrafanaOrganizationReconciler) reconcileCreate(ctx context.Context, graf
 	}
 
 	// Update CR status if anything was changed
-	if organization.ID != grafanaOrganization.Status.OrgID {
-		grafanaOrganization.Status.OrgID = organization.ID
-
-		if err = r.Status().Update(ctx, grafanaOrganization); err != nil {
-			logger.Error(err, "failed to update the status")
-			return ctrl.Result{}, errors.WithStack(err)
-		}
+	if err = r.updateOrgStatus(ctx, grafanaOrganization, organization.ID); err != nil {
+		return ctrl.Result{}, errors.WithStack(err)
 	}
 
-	// TODO add datasources for the organization.
-
 	err = r.configureGrafana(ctx)
 	if err != nil {
 		return ctrl.Result{}, errors.WithStack(err)
 	}
 	return ctrl.Result{}, nil
 }
 
+func (r GrafanaOrganizationReconciler) updateOrgStatus(ctx context.Context, grafanaOrganization *v1alpha1.GrafanaOrganization, orgID int64) error {
+	logger := log.FromContext(ctx)
+
+	// Update orgID in the CR's satus
+	if grafanaOrganization.Status.OrgID != orgID {
+		logger.Info("updating orgID in the org's status")
+		grafanaOrganization.Status.OrgID = orgID
+
+		if err := r.Status().Update(ctx, grafanaOrganization); err != nil {
+			logger.Error(err, "failed to update the status")
+			return errors.WithStack(err)
+		}
+	}
+
+	// Update the datasources in the CR's status
+	if grafanaOrganization.Status.DataSources == nil {
```

Comment:
I think you did not take the update into account? What if someone edited it?

Reply:
My assumption was that we are currently only adding the default `mimir` and `loki` datasources to each org and thus even if someone updates one of those 2, it won't change neither the orgID nor the org name which are the only things written in the CR status so it wouldn't make sense to update the CR's status.

However by writing this I realized that some users might want to update the datasource name which would entail a change in the CR status.So yeah, TL,DR we'll eventually need to take into account the update case.